elastic/kibana
# [ResponseOps] Implement a mechanism to copy source data into the alerts-as-data documents for ES Query rules

Resolves https://github.com/elastic/kibana/issues/168589

## Summary

This PR allows users to select _source fields that will then be copied into the alerts-as-data docs for the ES Query rule.
We will start by allowing them to choose from the following fields:
- host.name
- host.hostname
- host.id
- container.id
- kubernetes.pod.uid

<img width="468" alt="Screen Shot 2023-11-29 at 2 36 51 PM" src="https://github.com/elastic/kibana/assets/109488926/bd11b921-72be-4d84-8288-f4dca16c494e">

### Checklist


### To verify

1. Create or use an index that includes some of the fields listed above
Ex.

```
POST test-data/_doc/
{
  "@timestamp": "2023-11-27T10:00:00",
  "host": {
    "id": "1",
    "name": "host-1",
    "hostname": "host-1"
  }
}
```
2. Use that index to create ES Query rules with the different query types
3. Verify that only valid fields that exist in the index are visible in the combo box options
4. Verify that the valid fields are auto selected
5. Verify that if you change the index/dataview (or query for ESQL) that the selected fields are reset and and the combo box options reflect the query or index/dataview
6. Save the rule and verify that it runs and populates the new fields in the AAD documents
7. Create a new rule and use an index that doesn't contain any of the valid fields. Verify that the option to select fields is hidden
